spec/features/admin_node_update_feature_spec.rb

```diff
@@ -41,20 +41,30 @@
     end
     # rubocop:enable RSpec/MultipleExpectations
 
+    # rubocop:disable RSpec/ExampleLength
     scenario "User clicks on 'Reboot to update'", js: true do
-      allow(::Velum::Salt).to receive(:call)
+      allow(::Velum::Salt).to receive(:call).and_return(true)
 
       # clicks on "Update admin node"
       find(".update-admin-btn").click
 
+      # wait modal to appear
+      wait_until { page.has_text?("Reboot to update") }
+
       # clicks on "Reboot to update"
       find(".reboot-update-btn").click
 
-      # rubocop:disable RSpec/MessageSpies
-      expect(::Velum::Salt).to receive(:call)
-      # rubocop:enable RSpec/MessageSpies
+      wait_until { page.has_text?("Rebooting...") }
```

Comment:
A question from my side (I've looked it up in a quick search and it's not clear to me), is the `wait` time blocking ? That is, will it wait for 10 seconds and then perform the check ? or will it quit as soon as the expectation matches ?